app/src/main/java/de/danoeh/antennapod/ui/screen/playback/TranscriptFragment.java

```diff
@@ -0,0 +1,290 @@
+import android.view.ViewGroup;
+import android.view.WindowManager;
+import android.widget.ProgressBar;
+import android.widget.Toast;
+
+import com.google.android.material.dialog.MaterialAlertDialogBuilder;
+
+import de.danoeh.antennapod.databinding.TranscriptDialogBinding;
+import de.danoeh.antennapod.playback.base.PlayerStatus;
+import de.danoeh.antennapod.playback.service.PlaybackController;
+import de.danoeh.antennapod.ui.chapters.PodcastIndexTranscriptUtils;
+import org.greenrobot.eventbus.EventBus;
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.event.playback.PlaybackPositionEvent;
+import de.danoeh.antennapod.model.feed.FeedMedia;
+import de.danoeh.antennapod.model.feed.Transcript;
+import de.danoeh.antennapod.model.feed.TranscriptSegment;
+import de.danoeh.antennapod.model.playback.Playable;
+import de.danoeh.antennapod.ui.common.ThemeUtils;
+import de.danoeh.antennapod.ui.transcript.TranscriptViewholder;
+import io.reactivex.Maybe;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+
+public class TranscriptFragment extends DialogFragment {
+    public static final String TAG = "TranscriptFragment";
+    private TranscriptDialogBinding viewBinding;
+    private PlaybackController controller;
+    private Disposable disposable;
+    Playable media;
+    Transcript transcript;
+    TreeMap<Long, TranscriptSegment> segmentsMap;
+    private ProgressBar progressBar;
+    TranscriptAdapter adapter = null;
+    TranscriptViewholder currentView = null;
+    TranscriptViewholder prevView = null;
+    RecyclerView viewTranscript;
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    public TranscriptFragment initMedia(Playable m) {
+        media = m;
+        return this;
+    }
+
+    @Override
+    public void onResume() {
+        ViewGroup.LayoutParams params;
+        params = getDialog().getWindow().getAttributes();
+        params.width = WindowManager.LayoutParams.MATCH_PARENT;
+        getDialog().getWindow().setAttributes((WindowManager.LayoutParams) params);
+        getDialog().getWindow().setBackgroundDrawable(new ColorDrawable(Color.BLACK));
+        super.onResume();
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+        AlertDialog dialog = new MaterialAlertDialogBuilder(requireContext())
+                .setView(onCreateView(getLayoutInflater()))
+                .setPositiveButton(getString(R.string.close_label), null) //dismisses
+                .setTitle(R.string.transcript)
+                .create();
+        dialog.show();
+        dialog.getButton(DialogInterface.BUTTON_NEUTRAL).setVisibility(View.INVISIBLE);
```

Comment:
Should the neutral button have a text then? Currently it is invisible but can be clicked